test/importmaps_test.rb

```diff
@@ -0,0 +1,34 @@
+require "test_helper"
+
+class ImportmapsTest < ActiveSupport::TestCase
+  def setup
+    @importmaps = Importmap::Maps.new.tap do |maps|
+      maps.draw do
+        pin "application"
+
+        map :public, entry_point: 'public' do
+          pin "public", to: 'public/public.js'
+          pin "md5", to: "https://cdn.skypack.dev/md5", preload: true
+
+          pin_all_from "app/javascript/public/controllers", to: "public/controllers", under: "controllers", preload: true
+          pin_all_from "app/javascript/helpers", under: "helpers", preload: true
+        end
+      end
+    end
+  end
+
+  test "importmaps instance has a default map" do
+    h = generate_importmap_json('default')
+    assert_equal h['imports'].keys, [ 'application' ]
+  end
+
+  test "importmaps instance has an specific map" do
+    h = generate_importmap_json('public')
+    assert_equal h['imports'].keys, [ 'public', 'md5', 'controllers/hello_controller', 'controllers', 'helpers/requests' ]
```

Comment:
From what I understand going through the code and this line, this doesn't include the `"application"` part. I feel we should do this instead:

```rb
pin "whatever" # This should be included globally instead of only inside the `:default` map

map :public do
  pin "more_whatever" # This should be only included in the `:public` map
end
```

Basically, the above should generate:

```json
{
  "imports": {
    "whatever": "/assets/whatever.js",
    "more_whatever": "/assets/more_whatever.js"
  }
}
```

for the `:public` Import Map, and:

```json
{
  "imports": {
    "whatever": "/assets/whatever.js"
  }
}
```

for the `:default` Import Map.

This will allow not having to repeat a `pin`/`pin_all_from` if something should be included everywhere. Thoughts?